### PR TITLE
Port guide revamp

### PR DESCRIPTION
**Pull request by abkro**
Completed restructuring porting guide, including adding a section on apparmor kernel patching as well as a number of smaller edits. Added links to other resources from sections that have yet to be written. Corrected other links in the ubports docs so that they now point correctly to the new porting guide.

I am planning one or more new pull requests from this same branch once the text for the remaining sections is in place, so I will not be deleting this branch when my current changes have been merged.

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -163,4 +163,4 @@
 
 github_doc_root = ''
 def setup(app):
-    app.add_css_file('css/ubports.css')
+    app.add_stylesheet('css/ubports.css')
```
Comment (doniks) [inline conf.py:166]: Which version of sphinx and which OS are you using?

**Comment by abkro** [comment]
I've got Ubuntu 20.04. I'll check Sphinx version later. At work now.Ari21. sep. 2020 09:49 skrev doniks <notifications@github.com>:
@doniks commented on this pull request.



In conf.py:
> @@ -163,4 +163,4 @@
 
 github_doc_root = ''
 def setup(app):
-    app.add_css_file('css/ubports.css')
+    app.add_stylesheet('css/ubports.css')

Which version of sphinx and which OS are you using?

**Comment by abkro** [comment]
The two mentioned conflicting files have been deleted. I don't see how there can be a conflict? I did not find internal links in the rest of the guide targeting these pages either and my local build worked fine.

**Comment by doniks** [comment]
> The two mentioned conflicting files have been deleted. I don't see how there can be a conflict? I did not find internal links in the rest of the guide targeting these pages either and my local build worked fine.

i havent checked, but maybe these files have been changed after you created your branch. if that is the case, then step 1would be to doublr check that you havent lost/overwritten some contribution,and step 2 to rebase your branch on top

**Comment by abkro** [comment]
OK. I see what has happened here. I will correct later today and update here.

**Comment by abkro** [comment]
Retracting to fix before submitting new PR

**Pull request closed without merging**